### ISSUE TITLE
Improve default selection interactions on table

### ISSUE THREAD

**Issue by llorca** [palantir/blueprint#1079]
- [ ] Clicking on "..." in cell shouldn't select the cell, similarly to other "..." (header for instance)
- [ ] Column headers and row headers should reflect the current selection
- [ ] Clicking on a selected cell shouldn't deselect it. `cmd+click` on a cell should deselect it (#1073)

**Comment by samdenicola** [comment]
@llorca how about adding "onDoubleClick" to this list as talked about in #298?

**Comment by llorca** [comment]
@samdenicola Good catch. Can you file a different issue? This one is specifically related to selection--double click isn't about selection, it's just a brand new table interaction
